src/internal/partial_solution.rs

```diff
@@ -17,9 +17,26 @@ use crate::{
     solver::DependencyProvider,
 };
 
+#[derive(Debug, Copy, Clone, Ord, PartialOrd, Eq, PartialEq)]
+pub struct DecisionLevel(u32);
```

Comment:
Alright just in the following commit :+1: 
I feel like a type alias would be enough though.

Reply:
But ok for the newtype

Reply:
> Alright just in the following commit +1

Sorry, I don't follow what you mean here :)

Reply:
Just because I was reviewing one commit after the other. In the previous commit, I suggested giving a type alias for decision level and here in the following commit you created a type for that. Newtype is good :+1:

Reply:
Ah, and I misunderstood that you were arguing against adding a newtype or type alias for `decision_level` and wanted to keep a plain integer type. Got it now :+1: